configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java

```diff
@@ -110,17 +113,48 @@ public void decideVersion() {
         assertEquals(Vtag.currentVersion, ApplicationRepository.decideVersion(regularApp, Environment.perf, targetVersion));
     }
 
+    @Test
+    public void deleteUnusedFileReferences() {
+        File fileReferencesDir = Files.createTempDir();
```

Comment:
If this test fails before deletion (e.g. failing assert), the directory will be left on disk. This is likely the case for all uses of `Files.createTempDir()` below.

Consider using something like
```
@Rule
public TemporaryFolder folder = new TemporaryFolder();
```